scripts/src/public/auth.ts

```diff
@@ -15,3 +16,20 @@ export async function authenticate(req: express.Request): Promise<db.AuthToken>
 
 	return token;
 }
+
+export const authenticateUser = async function(req: express.Request, res: express.Response, next: express.NextFunction) {
+	const token = await authenticate(req);
+	httpContext.set("userId", token.userId);
+	httpContext.set("deviceId", token.deviceId);
+
+	const user = await db.User.findOneById(token.userId);
+
+	if (!user) {
+		// This error now defines an inconsistent state in the DB where a token exists but not user is found
+		// This should never happen as the token.user_id is a foreign key to the users table
+		throw TOSMissingOrOldToken();
+	}
+
+	req.context = { user, token };
```

Comment:
why is this needed if they are stored in the `httpContext`?

Reply:
They're not stored in the httpContext, but you're right. We could also push them into it

Reply:
ok